futures/tests/futures_unordered.rs

```diff
@@ -68,8 +67,8 @@ fn finished_future() {
     let (c_tx, c_rx) = oneshot::channel::<i32>();
 
     let mut stream = futures_unordered(vec![
-        FutureObj::new(Box::new(a_rx)),
-        //FutureObj::new(Box::new(b_rx.select(c_rx))),
+        a_rx.boxed() as BoxFuture<_>,
```

Comment:
hmm... I kind of think we should change the default return of `boxed` to the trait-object version to make this cast unnecessary. WDYT?

Reply:
Sounds good to me, if users want a non-trait-object they can call `Box::new` themselves.

Reply:
One issue with this is that `BoxFuture` can't actually be directly converted to `FutureObj` because of #1535, which is why https://github.com/rust-lang-nursery/futures-rs/pull/1494/files#diff-57ad78f21e9043ebe29659ff5d335774R9 was necessary. I've got a fix for #1535 ready to open once this is merged (it requires an enhancement to the `Pin` API so will be blocked on a `rust` PR).

Reply:
Sorry, can you clarify-- is #1535 blocked on this change, or is this change now blocked on #1535 ? What's the change to `Pin`? Do you have a PR open / issue open already for that?

Reply:
This change exacerbates the issue from #1535 since `FutureExt::boxed` now returns `BoxFuture` and `FutureObj::new` can’t take a `BoxFuture` since it doesn’t support boxed unsized values.

https://github.com/Nemo157/futures-rs/commit/b4361780faf764c7cb046ed75f863a6fcfd44800 is the change I’m referring to which will allow converting a `BoxFuture` to a `FutureObj`, that’s the one using the `Pin` api changes.

Reply:
Oh, the API changes I was referring to are https://github.com/rust-lang/rust/pull/60165, they can be worked around similar to the old code if we want to get a stable release out for 1.36.

Reply:
In that case, would you like to go ahead and land this change, or would you prefer to hold off?

Reply:
I think landing this as is is fine, I'll open a PR for that other change straight away so it can be merged once https://github.com/rust-lang/rust/pull/60165 is (or updated to work around it), I just didn't do it as a parallel PR as it would definitely have merge conflicts with this one.

Reply:
sg, merged this one.